TodoApp.ConsoleApp/Services/TodoService.cs

```diff
@@ -1,77 +1,87 @@
 ﻿using System.Collections.Generic;
-using TodoApp.Library.Data;
-using TodoApp.Library.Models;
+using TodoApp.ConsoleApp.Repositories.Models;
+using TodoApp.ConsoleApp.Repositories.XMLRepository;
 using System.Linq;
 using System;
+using System.IO;
+using System.Xml;
+using System.Text;
 
 namespace TodoApp.ConsoleApp.Services
 {
     public class TodoService : ITodoService
     {
-        public IEnumerable<Task> GetAll()
-        {
-            XMLTaskReader reader = new XMLTaskReader();
-
-            IEnumerable<Task> tasks = reader.ReadTasks();
+        static string Path { get; set; }
+        TodoRepository repo = new TodoRepository(Path);
 
-            return tasks;
+        public IEnumerable<TodoModel> GetAll()
+        {
+            return repo.GetAll();
         }
 
-        public Task GetByID(Guid id)
+        public TodoModel GetByID(Guid id)
         {
-            IEnumerable<Task> tasks = GetAll();
-
-            Task wantedTask = null;
-
-            wantedTask = tasks.FirstOrDefault(task => task.ID == id);
-
-            return wantedTask;
+            return repo.GetById(id);
         }
 
-        public bool Update(Task task)
+        public bool Update(TodoModel model)
         {
-            if (task == null)
+            if (model == null)
             {
-                throw new ArgumentNullException(nameof(task));
+                throw new ArgumentNullException(nameof(model));
             }
 
-            bool isDeleted = Delete(task.ID);
+            bool isDeleted = Delete(model.ID);
 
             if (!isDeleted)
             {
                 return false;
             }
 
-            var newTask = Create(task);
+            var newModel = Create(model);
 
-            return newTask != null;
+            return newModel != null;
         }
 
-        public Task Create(Task task)
+        public TodoModel Create(TodoModel model)
```

Comment:
We need to return the **result** from `Insert()`
The old model do not have ID.
We do not need the first part of the method anymore.